org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/util/JRTUtil.java

```diff
@@ -477,7 +477,7 @@ boolean sameRelease(String other) {
 
 	static Path toJdkHome(File jrt) {
 		Path home;
-		Path normalized = jrt.toPath().normalize();
+		Path normalized = jrt.toPath().toAbsolutePath().normalize();
```

Comment:
When you need to introduce such Filesystem calls please provide measurement. Some of these Path method look harmless but system dependently do much more then they look like. May be its fast - may be not. It may even differ by the provided path and used filesystem. 
I am sure its faster then toRealPath but fast enough for this usecase? Does it really "operate without actually accessing the file-system."?
And why do you even introduce toAbsolutePath() here when it was not needed before?

Reply:
> When you need to introduce such Filesystem calls please provide measurement. Some of these Path method look harmless but system dependently do much more then they look like. May be its fast - may be not. It may even differ by the provided path and used filesystem.

Sorry, but I can't provide such extensive testing. I only have a Windows computer available and not the time for a test-campaign.

> I am sure its faster then toRealPath but fast enough for this usecase? Does it really "operate without actually accessing the file-system."?

When looking at the `WindowsPath` implementation I can only see String operations. And when looking at the doc, while `Path.toRealPath()` requires the file to exist and may throw an `IOException` `toAbsolutePath()` does have this requirement and does not throw `IOExceptions`. At the same time the latter method is a NOP if the path is already absolute (which should be the case usually).
While I cannot guarantee that `toAbsolutePath()` does not access the file-system under all circumstances, I'm relatively confident that it does not at least in most cases.

For `Path.normalize()` the JavaDoc even mentions it explicitly:
`This method does not access the file system; the path may not locate a file that exists.`


> And why do you even introduce toAbsolutePath() here when it was not needed before?

In some places it was already used as you can see in https://github.com/eclipse-jdt/eclipse.jdt.core/pull/2371, some used `toAbsolutePath()` (respectively the File equivalent) and as you can see in this PR some used `normalized()` and because the created objects are retained for the entire lifetime of a JVM it actual duplicates should be avoided.